pht_federated/aggregator/clients/discovery_client.py

```diff
@@ -0,0 +1,97 @@
+import requests
+import os
+from typing import Union, List
+from uuid import uuid4
+from requests.models import Response
+
+class DiscoveryClient:
```

Comment:
all methods should return the correct pydantic models. add error handling (`r.raise_for_status()`) for the requests